tests/Test/Pager/Pagination/PaginatorTest.php

```diff
@@ -2,22 +2,31 @@
 
 namespace Test\Pager\Pagination;
 
+use Knp\Component\Pager\Exception\PageLimitInvalidException;
+use Knp\Component\Pager\Exception\PageNumberInvalidException;
 use Test\Tool\BaseTestCase;
 
 final class PaginatorTest extends BaseTestCase
 {
     /**
      * @test
      */
-    public function shouldThrowExceptionOnInvalidPageAndLimitArgs(): void
+    public function shouldThrowExceptionIfPageIsInvalid(): void
     {
-        $exceptionThrown = false;
-        try {
-            $paginator = $this->getPaginatorInstance();
-            $paginator->paginate(['a', 'b'], -1, 0);
-        } catch (\LogicException $e) {
-            $exceptionThrown = true;
-        }
-        $this->assertTrue($exceptionThrown);
+        $this->expectException(PageNumberInvalidException::class);
+
+        $paginator = $this->getPaginatorInstance();
+        $paginator->paginate(['a', 'b'], 0, -1);
```

Comment:
Can we use a valid value for limit here? I know that it's just the same, because the exception are thrown in a certain order, but it's just to be clearer